tests/basic/tests.py

```diff
@@ -367,6 +367,7 @@ def test_emptyqs(self):
         with self.assertRaises(TypeError):
             EmptyQuerySet()
         self.assertIsInstance(Article.objects.none(), EmptyQuerySet)
+        self.assertFalse(isinstance(Article.objects.all(), EmptyQuerySet))
```

Comment:
This assertion won't fail without the provided fix since `Article.objects.all()` is an instance of `QuerySet` which has a `query` attribute. You should pass a non-`QuerySet` object to `isinstance` instead.